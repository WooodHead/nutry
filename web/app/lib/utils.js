import _ from 'lodash';

function interpolateColor(color1, color2, factor) {
  if (arguments.length < 3) {
    factor = 0.5;
  }
  const result = color1.slice();
  for (let i = 0; i < 3; i += 1) {
    result[i] = Math.round(result[i] + (factor * (color2[i] - color1[i])));
  }
  return convertToHex(result);
}
export function interpolateColors(color1, color2, steps) {
  const stepFactor = 1 / (steps - 1),
    interpolatedColorArray = [];

  color1 = color1.match(/\d+/g).map(Number);
  color2 = color2.match(/\d+/g).map(Number);

  for (let i = 0; i < steps; i++) {
    interpolatedColorArray.push(interpolateColor(color1, color2, stepFactor * i));
  }

  return interpolatedColorArray;
}

function hex(c) {
  const s = '0123456789abcdef';
  let i = parseInt(c);
  if (i === 0 || isNaN(c)) { return '00'; }
  i = Math.round(Math.min(Math.max(0, i), 255));
  return s.charAt((i - (i % 16)) / 16) + s.charAt(i % 16);
}

/* Convert an RGB triplet to a hex string */
function convertToHex(rgb) {
  return hex(rgb[0]) + hex(rgb[1]) + hex(rgb[2]);
}

/* Remove '#' in color hex string */
function trim(s) { return (s.charAt(0) === '#') ? s.substring(1, 7) : s; }

/* Convert a hex string to an RGB triplet */
function convertToRGB(hexString) {
  const color = [];
  color[0] = parseInt((trim(hexString)).substring(0, 2), 16);
  color[1] = parseInt((trim(hexString)).substring(2, 4), 16);
  color[2] = parseInt((trim(hexString)).substring(4, 6), 16);
  return color;
}
export function generateColor(colorStart, colorEnd, colorCount) {
  // The beginning of your gradient
  const start = convertToRGB(colorStart);

  // The end of your gradient
  const end = convertToRGB(colorEnd);

  // The number of colors to compute
  const len = colorCount;

  // Alpha blending amount
  let alpha = 0.0;

  const saida = [];

  for (let i = 0; i < len; i += 1) {
    const c = [];
    alpha += (1.0 / len);

    c[0] = (start[0] * alpha) + ((1 - alpha) * end[0]);
    c[1] = (start[1] * alpha) + ((1 - alpha) * end[1]);
    c[2] = (start[2] * alpha) + ((1 - alpha) * end[2]);

    saida.push(`#${convertToHex(c)}`);
  }

  return saida;
}


export const arrayToObject = (array) =>
   array.reduce((obj, item) => {
     obj[item.id] = item;
     return obj;
   }, {});


export function scaledValue(value, portion = false) {
  if (!portion) {
    return truncateTo(value, 2);
  } else if (value) {
    return truncateTo(value * scale(portion), 2);
  }

  return 0;
}

const truncateTo = (unRouned, nrOfDecimals = 2) => {
  const parts = String(unRouned).split('.');

  if (parts.length !== 2) {
          // without any decimal part
    return unRouned;
  }

  const newDecimals = parts[1].slice(0, nrOfDecimals);
  const newString = `${parts[0]}.${newDecimals}`;
  return Number(newString);
};

function scale(portion) {
    /**
     * Attempt to parse combinations of scales/portions and return an numerical
     * value for usage by higher order functions.
     */
  const defaultUnit = 100; // per 100g is the default measurement

  if (_.isInteger(portion)) {
      // portion = 200 (g)
    return _.parseInt(portion) / defaultUnit;
  } else if (isFloat(portion)) {
      // portion = 28.5 (g)
    return parseFloat(portion) / defaultUnit;
  } else if (!isNumeric(portion)) {
      // portion = 1 tsp
  } else {
    return 1;
  }
  return 1;
}
function isFloat(n) {
  return isNumeric(n) && !_.isInteger(n);
}
function isNumeric(n) {
  return !_.isNaN(n) && _.isFinite(n);
}
