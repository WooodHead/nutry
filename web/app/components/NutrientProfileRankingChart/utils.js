import React from 'react';
import PropTypes from 'prop-types';

import { generateColor, scaledValue } from 'lib/utils';
import { prefixToName, prefixToUnit } from 'lib/nutrientMap';

export const CustomTooltip = (props) => {
  const { active, nutrientSelected, portionSelected } = props;
  if (active) {
    const { payload, foodID } = props;
    if (payload) {
      const { name, value, unit, id } = payload ? payload[0].payload : null;
      // TODO: add description mapping
      return (
        <div className="custom-tooltip">
          <p className="label">{foodID === id ? `${name} [Current FoodProfile]` : `${name}`}</p>
          <p className="intro">{`Contains ${value} ${unit} of ${nutrientSelected} per ${portionSelected.unit}`}</p>
          <p className="desc">Description: Coming soon!</p>
        </div>
      );
    }
    return (
      <div className="custom-tooltip">
      </div>
    );
  }
  return null;
};
CustomTooltip.propTypes = {
  payload: PropTypes.array,
};
export const CustomShape = (props) => { /* eslint react/prop-types: 0 */
  const barId = props.payload.id;
  const { fillActive, fill, x, y, width, height, radius, foodID } = props;
  // console.log(props);
  return <path d={getPath(x, y, width, height, radius)} stroke="none" fill={barId === foodID ? fillActive : fill} />;
};
const getPath = (x, y, width, height, radius) => {
  const maxRadius = Math.min(Math.abs(width) / 2, Math.abs(height) / 2);
  const sign = height >= 0 ? 1 : -1;
  const clockWise = height >= 0 ? 1 : 0;
  let path;

  if (maxRadius > 0 && radius instanceof Array) {
    const newRadius = [];
    for (let i = 0, len = 4; i < len; i += 1) {
      newRadius[i] = radius[i] > maxRadius ? maxRadius : radius[i];
    }

    path = `M${x},${y + (sign * newRadius[0])}`;

    if (newRadius[0] > 0) {
      path += `A ${newRadius[0]},${newRadius[0]},0,0,${clockWise},${x + newRadius[0]},${y}`;
    }

    path += `L ${(x + width) - newRadius[1]},${y}`;

    if (newRadius[1] > 0) {
      path += `A ${newRadius[1]},${newRadius[1]},0,0,${clockWise},
       ${x + width},${y + (sign * newRadius[1])}`;
    }
    path += `L ${x + width},${(y + height) - (sign * newRadius[2])}`;

    if (newRadius[2] > 0) {
      path += `A ${newRadius[2]},${newRadius[2]},0,0,${clockWise},
       ${(x + width) - newRadius[2]},${y + height}`;
    }
    path += `L ${x + newRadius[3]},${y + height}`;

    if (newRadius[3] > 0) {
      path += `A ${newRadius[3]},${newRadius[3]},0,0,${clockWise},
       ${x},${(y + height) - (sign * newRadius[3])}`;
    }
    path += 'Z';
  } else if (maxRadius > 0 && radius === +radius && radius > 0) {
    const newRadius = Math.min(maxRadius, radius);

    path = `M ${x},${y + (sign * newRadius)}
           A ${newRadius},${newRadius},0,0,${clockWise},${x + newRadius},${y}
           L ${x + (width - newRadius)},${y}
           A ${newRadius},${newRadius},0,0,${clockWise},${x + width},${y + (sign * newRadius)}
           L ${x + width},${(y + height) - (sign * newRadius)}
           A ${newRadius},${newRadius},0,0,${clockWise},${x + (width - newRadius)},${y + height}
           L ${x + newRadius},${y + height}
           A ${newRadius},${newRadius},0,0,${clockWise},${x},${(y + height) - (sign * newRadius)} Z`;
  } else {
    path = `M ${x},${y} h ${width} v ${height} h ${-width} Z`;
  }
  return path;
};

export function processData(rawData, nutrientSelected, portionSelected) {
  if (rawData.length === 0) {
    return [];
  }
  const barData = rawData.get(nutrientSelected);
  if (!barData) {
    return [];
  }
  const COLORS = generateColor('#A75FFF', '#5A338A', barData.length);
  return barData
  .filter((value) => value.get('value') !== '~')
      .sort((a, b) => b.get('value') - a.get('value'))
        .map((value, index) => value.set('fill', COLORS[index % COLORS.length]).set('value', typeof value.get('value') === 'number' ? scaledValue(value.get('value'), portionSelected.g) : 0).toJS());
}

export const xLabel = (nutrient) => `${prefixToName(nutrient)} [${prefixToUnit(nutrient)}]`;
