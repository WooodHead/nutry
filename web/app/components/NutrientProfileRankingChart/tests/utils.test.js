import { Map } from 'immutable';
import { CustomShape, CustomTooltip, processData, xLabel } from '../utils';

describe('utils', () => {
  describe('CustomShape()', () => {
    it('renders correct custom shape', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: 45,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape different radius', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: 45,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
        radius: 10,
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape radius array', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: 45,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
        radius: [1, 2, 3, 4],
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape radius array that is negative', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: 45,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
        radius: [-1, -1, -1, -1],
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape radius array is large', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: 45,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
        radius: [50, 50, 50, 50],
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape negative height', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: -1,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
        radius: [-1, -1, -1, -1],
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape negative height', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: -1,
        name: 'braised beef thymus',
        id: '13338',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13338', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
        radius: [-1, -1, -1, -1],
      };
      const customShape = CustomShape(props);
      expect(customShape).toMatchSnapshot();
    });
    it('renders correct custom shape when barId == foodID (i.e active)', () => {
      const props = {
        fillActive: '#50c59f',
        foodID: '13337',
        width: 220.2362304194038,
        height: 45,
        name: 'braised beef thymus',
        id: '13337',
        value: 52.83,
        unit: 'g',
        fill: '#9a57eb',
        x: 140,
        y: 312.3333333333333,
        payload: { name: 'braised beef thymus', id: '13337', value: 52.83, unit: 'g', fill: '#9a57eb' },
        background: { x: 140, y: 312.3333333333333, width: 330, height: 45 },
        index: 4,
      };
      const customShape = CustomShape(props);
      expect(customShape.props.fill).toEqual('#50c59f');
      expect(customShape).toMatchSnapshot();
    });
  });
  describe('CustomTooltip()', () => {
    it('renders correct tooltip', () => {
      const props = {
        active: true,
        nutrientSelected: 'Protien',
        portionSelected: { unit: 'per 100g', value: 100, className: 'per100g', label: 'per 100g', amt: 1 },
        foodID: '13337',
        payload: [
          {
            color: null,
            dataKey: 'value',
            name: 'value',
            payload: {
              fill: '#663a9d',
              id: '17219',
              name: 'braised veal thymus',
              unit: 'g',
              value: 22.67,
            },
          },
        ],
      };
      const customTooltip = CustomTooltip(props);
      expect(customTooltip).toMatchSnapshot();
    });
    it('renders when no info in payload', () => {
      const props = {
        active: true,
        nutrientSelected: 'Protien',
        portionSelected: { unit: 'per 100g', value: 100, className: 'per100g', label: 'per 100g', amt: 1 },
        foodID: '13337',
        payload: [
          {
            color: null,
            dataKey: 'value',
            name: 'value',
            payload: {
              fill: null,
              id: null,
              name: null,
              unit: null,
              value: null,
            },
          },
        ],
      };
      const customTooltip = CustomTooltip(props);
      expect(customTooltip).toMatchSnapshot();
    });
    it('renders correct tooltip when foodId == id', () => {
      const props = {
        active: true,
        nutrientSelected: 'Protien',
        portionSelected: { unit: 'per 100g', value: 100, className: 'per100g', label: 'per 100g', amt: 1 },
        foodID: '13337',
        id: '13337',
        payload: [
          {
            color: null,
            dataKey: 'value',
            name: 'value',
            payload: {
              fill: '#663a9d',
              id: '13337',
              name: 'braised veal thymus',
              unit: 'g',
              value: 22.67,
            },
          },
        ],
      };
      const customTooltip = CustomTooltip(props);
      expect(customTooltip).toMatchSnapshot();
    });
    it('renders correct when no payload', () => {
      const props = {
        active: true,
        nutrientSelected: 'Protien',
        portionSelected: { unit: 'per 100g', value: 100, className: 'per100g', label: 'per 100g', amt: 1 },
        foodID: '13337',
        id: '13337',
        payload: null,
      };
      const customTooltip = CustomTooltip(props);
      expect(customTooltip).toMatchSnapshot();
    });
    it('renders correct when no active=false', () => {
      const props = {
        active: false,
        nutrientSelected: 'Protien',
        portionSelected: { unit: 'per 100g', value: 100, className: 'per100g', label: 'per 100g', amt: 1 },
        foodID: '13337',
        id: '13337',
        payload: null,
      };
      const customTooltip = CustomTooltip(props);
      expect(customTooltip).toEqual(null);
    });
  });
  describe('xLabel()', () => {
    it('xLabel is correct', () => {
      const nutrient = { prefix: 'WATER', value: 52.83, unit: 'g' };
      const xLabelString = xLabel(nutrient.prefix);
      expect(xLabelString).toEqual('Water [g]');
    // console.log(activeShape.props.children[6]);
    });
  });
  describe('processData()', () => {
    it('handles rawdata undefined', () => {
      const data = null;
      const nutrientSelected = null;
      const portionSelected = null;
      expect(processData(data, nutrientSelected, portionSelected)).toEqual([]);
    // console.log(activeShape.props.children[6]);
    });
    it('handles no raw data', () => {
      const data = [];
      const nutrientSelected = null;
      const portionSelected = null;
      expect(processData(data, nutrientSelected, portionSelected)).toEqual([]);
    // console.log(activeShape.props.children[6]);
    });
    it('handles no raw data', () => {
      const data = Map({
        ALA: [],
        ARG: [],
        CA: [
          Map({ name: 'raw beef thymus', id: '13337', value: 7, unit: 'mg' }),
          Map({ name: 'braised beef thymus', id: '13338', value: 10, unit: 'mg' }),
          Map({ name: 'Raw veal thymus', id: '17218', value: 3, unit: 'mg' }),
          Map({ name: 'braised veal thymus', id: '17219', value: 4, unit: 'mg' }),
          Map({ name: 'Fresh thyme', id: '02049', value: 405, unit: 'mg' }),
          Map({ name: 'Dried thyme', id: '02042', value: 1890, unit: 'mg' }),
        ],
      });
      const nutrientSelected = 'CHOCOLATE';
      const portionSelected = null;
      expect(processData(data, nutrientSelected, portionSelected)).toEqual([]);
    });
    it('handles processing data when found', () => {
      const data = Map({
        ALA: [],
        ARG: [],
        CA: [
          Map({ name: 'raw beef thymus', id: '13337', value: 7, unit: 'mg' }),
          Map({ name: 'braised beef thymus', id: '13338', value: 10, unit: 'mg' }),
          Map({ name: 'Raw veal thymus', id: '17218', value: 3, unit: 'mg' }),
          Map({ name: 'braised veal thymus', id: '17219', value: 4, unit: 'mg' }),
          Map({ name: 'Fresh thyme', id: '02049', value: 405, unit: 'mg' }),
          Map({ name: 'Dried thyme', id: '02042', value: 1890, unit: 'mg' }),
        ],
      });
      const nutrientSelected = 'CA';
      const portionSelected = {
        amt: 1,
        className: 'per100g',
        g: 100,
        label: 'per 100g',
        unit: 'per 100g',
        value: 100,
      };
      expect(processData(data, nutrientSelected, portionSelected).length).toEqual(6);
    });
    it('handles processing data nutrient has non number value', () => {
      const data = Map({
        ALA: [],
        ARG: [],
        CA: [
          Map({ name: 'raw beef thymus', id: '13337', value: 'NA', unit: 'mg' }),
          Map({ name: 'braised beef thymus', id: '13338', value: 'NA', unit: 'mg' }),
          Map({ name: 'Raw veal thymus', id: '17218', value: 'NA', unit: 'mg' }),
          Map({ name: 'braised veal thymus', id: '17219', value: 4, unit: 'mg' }),
          Map({ name: 'Fresh thyme', id: '02049', value: 405, unit: 'mg' }),
          Map({ name: 'Dried thyme', id: '02042', value: 1890, unit: 'mg' }),
        ],
      });
      const nutrientSelected = 'CA';
      const portionSelected = {
        amt: 1,
        className: 'per100g',
        g: 100,
        label: 'per 100g',
        unit: 'per 100g',
        value: 100,
      };
      expect(processData(data, nutrientSelected, portionSelected).length).toEqual(6);
    });
  });
});
