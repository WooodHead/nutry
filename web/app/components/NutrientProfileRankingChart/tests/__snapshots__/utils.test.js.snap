// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils CustomShape() renders correct custom shape 1`] = `
<path
  d="M 140,312.3333333333333 h 220.2362304194038 v 45 h -220.2362304194038 Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape different radius 1`] = `
<path
  d="M 140,322.3333333333333
           A 10,10,0,0,1,150,312.3333333333333
           L 350.2362304194038,312.3333333333333
           A 10,10,0,0,1,360.2362304194038,322.3333333333333
           L 360.2362304194038,347.3333333333333
           A 10,10,0,0,1,350.2362304194038,357.3333333333333
           L 150,357.3333333333333
           A 10,10,0,0,1,140,347.3333333333333 Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape negative height 1`] = `
<path
  d="M140,313.3333333333333L 361.2362304194038,312.3333333333333L 360.2362304194038,310.3333333333333L 139,311.3333333333333Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape negative height 2`] = `
<path
  d="M140,313.3333333333333L 361.2362304194038,312.3333333333333L 360.2362304194038,310.3333333333333L 139,311.3333333333333Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape radius array 1`] = `
<path
  d="M140,313.3333333333333A 1,1,0,0,1,141,312.3333333333333L 358.2362304194038,312.3333333333333A 2,2,0,0,1,
       360.2362304194038,314.3333333333333L 360.2362304194038,354.3333333333333A 3,3,0,0,1,
       357.2362304194038,357.3333333333333L 144,357.3333333333333A 4,4,0,0,1,
       140,353.3333333333333Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape radius array is large 1`] = `
<path
  d="M140,334.8333333333333A 22.5,22.5,0,0,1,162.5,312.3333333333333L 337.7362304194038,312.3333333333333A 22.5,22.5,0,0,1,
       360.2362304194038,334.8333333333333L 360.2362304194038,334.8333333333333A 22.5,22.5,0,0,1,
       337.7362304194038,357.3333333333333L 162.5,357.3333333333333A 22.5,22.5,0,0,1,
       140,334.8333333333333Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape radius array that is negative 1`] = `
<path
  d="M140,311.3333333333333L 361.2362304194038,312.3333333333333L 360.2362304194038,358.3333333333333L 139,357.3333333333333Z"
  fill="#9a57eb"
  stroke="none"
/>
`;

exports[`utils CustomShape() renders correct custom shape when barId == foodID (i.e active) 1`] = `
<path
  d="M 140,312.3333333333333 h 220.2362304194038 v 45 h -220.2362304194038 Z"
  fill="#50c59f"
  stroke="none"
/>
`;

exports[`utils CustomTooltip() renders correct tooltip 1`] = `
<div
  className="custom-tooltip"
>
  <p
    className="label"
  >
    braised veal thymus
  </p>
  <p
    className="intro"
  >
    Contains 22.67 g of Protien per per 100g
  </p>
  <p
    className="desc"
  >
    Description: Coming soon!
  </p>
</div>
`;

exports[`utils CustomTooltip() renders correct tooltip when foodId == id 1`] = `
<div
  className="custom-tooltip"
>
  <p
    className="label"
  >
    braised veal thymus [Current FoodProfile]
  </p>
  <p
    className="intro"
  >
    Contains 22.67 g of Protien per per 100g
  </p>
  <p
    className="desc"
  >
    Description: Coming soon!
  </p>
</div>
`;

exports[`utils CustomTooltip() renders correct when no payload 1`] = `
<div
  className="custom-tooltip"
/>
`;

exports[`utils CustomTooltip() renders when no info in payload 1`] = `
<div
  className="custom-tooltip"
>
  <p
    className="label"
  >
    NA
  </p>
  <p
    className="intro"
  >
    Contains NA NA of Protien per per 100g
  </p>
  <p
    className="desc"
  >
    Description: Coming soon!
  </p>
</div>
`;
